### PR TITLE
[Merged by Bors] - Chore(Algebra/Polynomial/Module/Basic): Refactor out Module.AEval 

### PR DESCRIPTION
**Pull request by Shamrock-Frost**
Refactor Module.AEval out of Algebra/Polynomial/Module/Basic.

---

See #13573 for previous discussion

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Polynomial/Module/AEval.lean [hunk 1]
```diff
@@ -0,0 +1,197 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
```
Comment (erdOne) [inline Mathlib/Algebra/Polynomial/Module/AEval.lean:4]: This is actually done by Richard in #7728.
```suggestion
Copyright (c) 2022 Richard M. Hill. All rights reserved.
Released under Apache 2.0 license as described in the file LICENSE.
Authors: Richard M. Hill
```

**Comment by erdOne** [comment]
LGTM otherwise. Thanks!

File: Mathlib/Algebra/Polynomial/Module/AEval.lean (hunk 1 above)
Comment (Shamrock-Frost) [reply Mathlib/Algebra/Polynomial/Module/AEval.lean]: good catch!

**Comment by erdOne** [comment]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**